src/math_utils.rs

```diff
@@ -69,10 +89,10 @@ fn igcdex(num_a: BigInt, num_b: BigInt) -> (BigInt, BigInt, BigInt) {
     }
 }
 ///Finds a nonnegative integer x < p such that (m * x) % p == n.
-fn div_mod(n: BigInt, m: BigInt, p: BigInt) -> BigInt {
-    let (a, _, c) = igcdex(m, p.clone());
-    assert_eq!(c, bigint!(1));
-    (n * a).mod_floor(&p)
+pub fn div_mod(n: &BigInt, m: &BigInt, p: &BigInt) -> BigInt {
+    let (a, _, c) = igcdex(m.clone(), p.clone());
```

Comment:
No need for clones, let's just use refs.

Reply:
resolved! :D
https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/0d96379ce5e302e39d889883238fbdbb97e48486